dds/OpenDDSConfigWrapper.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DDS_OPENDDSCONFIG_WRAPPER_H
+#define OPENDDS_DDS_OPENDDSCONFIG_WRAPPER_H
+
+// OpenDDSConfig.h is generated by the configure script or by CMake
+// If you're manually configuring OpenDDS, start by creating an empty file
+// named OpenDDSConfig.h in a dds/ directory on the include path.
+// See the OpenDDS Developer's Guide and dds/OpenDDSConfig.h.in for details
+// regarding what can be customized.
+#ifndef OPENDDS_IGNORE_OPENDDSCONFIG_H_FILE
+#  include <dds/OpenDDSConfig.h>
```

Comment:
Looks like the IDL compilers needs to `-I` the directory for this in CMake. It's generated here:

https://github.com/OpenDDS/OpenDDS/blob/85561d54d10af95f6ff05b06e34a279983550600/CMakeLists.txt#L62-L81

Possible options:

- Maybe if we had #4432, we could use it on the `OpenDDS_Config` target. I have the start of include inheriting as a prototype, but it needs a bit more design work, so it probably isn't viable for this PR.

- We could add `OPENDDS_IDL_OPTIONS -I${config_out_dir} TAO_IDL_OPTIONS -I${config_out_dir}` to the `opendds_target_sources` calls that need the config header.

- We just assume all IDL files should be able to include this and set `config_out_dir` to a cache variable that `opendds_target_sources` checks for and includes it. I think this one's the one I favor.

Reply:
https://github.com/OpenDDS/OpenDDS/blob/85561d54d10af95f6ff05b06e34a279983550600/cmake/opendds_target_sources.cmake#L387

Reply:
https://github.com/OpenDDS/OpenDDS/blob/85561d54d10af95f6ff05b06e34a279983550600/cmake/opendds_target_sources.cmake#L353

Actual place were the check should go.